presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanChecker.java

```diff
@@ -77,26 +91,95 @@ public PlanChecker(FeaturesConfig featuresConfig, boolean forceSingleNode)
 
     public void validateFinalPlan(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
     {
+        ensureProvidedInitialized();
         checkers.get(Stage.FINAL).forEach(checker -> checker.validate(planNode, session, metadata, warningCollector));
     }
 
     public void validateIntermediatePlan(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
     {
+        ensureProvidedInitialized();
         checkers.get(Stage.INTERMEDIATE).forEach(checker -> checker.validate(planNode, session, metadata, warningCollector));
     }
 
-    public void validatePlanFragment(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
+    public void validatePlanFragment(PlanFragment planFragment, Session session, Metadata metadata, WarningCollector warningCollector)
+    {
+        ensureProvidedInitialized();
+        checkers.get(Stage.FRAGMENT).forEach(checker -> checker.validateFragment(planFragment, session, metadata, warningCollector));
+    }
+
+    private static List<Checker> fromSpi(List<com.facebook.presto.spi.plan.PlanChecker> checkers)
```

Comment:
call this `fromSpiPlanChecker()` to make it clearer what this does, but if we call the plugin plan checkers separately, we won't need this adapter.

Reply:
this is removed now